include/wx/uwp/control.h

```diff
@@ -0,0 +1,136 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        wx/uwp/control.h
+// Purpose:     wxControl class
+// Author:      Yann Clotioloman Yéo
+// Modified by:
+// Created:     02/01/2022
+// Copyright:   (c) Julian Smart
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_CONTROL_H_
+#define _WX_CONTROL_H_
+
+#include "wx/dynarray.h"
+
+// General item class
+class WXDLLIMPEXP_CORE wxControl : public wxControlBase
+{
+public:
+    wxControl() { }
+
+    wxControl(wxWindow *parent, wxWindowID id,
+              const wxPoint& pos = wxDefaultPosition,
+              const wxSize& size = wxDefaultSize, long style = 0,
+              const wxValidator& validator = wxDefaultValidator,
+              const wxString& name = wxASCII_STR(wxControlNameStr))
+    {
+        Create(parent, id, pos, size, style, validator, name);
+    }
+
+    bool Create(wxWindow *parent, wxWindowID id,
+            const wxPoint& pos = wxDefaultPosition,
+            const wxSize& size = wxDefaultSize, long style = 0,
+            const wxValidator& validator = wxDefaultValidator,
+            const wxString& name = wxASCII_STR(wxControlNameStr));
+
+
+    // Simulates an event
+    virtual void Command(wxCommandEvent& event) wxOVERRIDE { ProcessCommand(event); }
+
+
+    // implementation from now on
+    // --------------------------
+
+    virtual wxVisualAttributes GetDefaultAttributes() const wxOVERRIDE
+    {
+        return GetClassDefaultAttributes(GetWindowVariant());
+    }
+
+    static wxVisualAttributes
+    GetClassDefaultAttributes(wxWindowVariant variant = wxWINDOW_VARIANT_NORMAL);
+
+    // Calls the callback and appropriate event handlers
+    bool ProcessCommand(wxCommandEvent& event);
+
+    // MSW-specific
+    virtual bool MSWOnNotify(int idCtrl, WXLPARAM lParam, WXLPARAM *result) wxOVERRIDE;
+
+    // For ownerdraw items
+    virtual bool MSWOnDraw(WXDRAWITEMSTRUCT *WXUNUSED(item)) { return false; }
+    virtual bool MSWOnMeasure(WXMEASUREITEMSTRUCT *WXUNUSED(item)) { return false; }
+
+    const wxArrayLong& GetSubcontrols() const { return m_subControls; }
+
+    // default handling of WM_CTLCOLORxxx: this is public so that wxWindow
+    // could call it
+    virtual WXHBRUSH MSWControlColor(WXHDC pDC, WXHWND hWnd);
+
+    // default style for the control include WS_TABSTOP if it AcceptsFocus()
+    virtual WXDWORD MSWGetStyle(long style, WXDWORD *exstyle) const wxOVERRIDE;
```

Comment:
I'm pretty sure we shouldn't have all this wxMSW stuff here

Reply:
Yes they will be removed :)